File: app/src/main/java/com/boardgamegeek/ui/model/GeekList.java
package com.boardgamegeek.ui.model;

import com.google.auto.value.AutoValue;

@AutoValue
public abstract class GeekList {
	public abstract int id();

	public abstract String title();

	public abstract String username();

	public abstract int numberOfItems();

	public abstract int numberOfThumbs();

	public abstract String description();

	public abstract long postTicks();

	public abstract long editTicks();

	public static GeekList.Builder builder() {
		return new AutoValue_GeekList.Builder();
	}

	@AutoValue.Builder
	public abstract static class Builder {
		public abstract GeekList.Builder setId(int value);

		public abstract GeekList.Builder setTitle(String value);

		public abstract GeekList.Builder setUsername(String value);

		public abstract GeekList.Builder setNumberOfItems(int value);

		public abstract GeekList.Builder setNumberOfThumbs(int value);

		public abstract GeekList.Builder setDescription(String value);

		public abstract GeekList.Builder setPostTicks(long value);

		public abstract GeekList.Builder setEditTicks(long value);

		public abstract GeekList build();
	}
}

File: app/src/main/java/com/boardgamegeek/ui/model/GeekList.kt
package com.boardgamegeek.ui.model

import com.boardgamegeek.model.GeekListItem

data class GeekList(
        val id: Int,
        val title: String,
        val username: String,
        val description: String,
        val numberOfItems: Int,
        val numberOfThumbs: Int,
        val postTicks: Long,
        val editTicks: Long,
        val items: List<GeekListItem>
)
